### PR TITLE
Update user active boolean with psu_identity gem

### LINKED ISSUE #843: Rescue Mailer errors that aren't recoverable
# Bug Report

## Description
The monthly email job will retry sending to an unknown user, and then eventually end up in the deadset queue. we should perhaps rescue these errros

## To Reproduce
```
Net::SMTPFatalError: 550 5.1.1 <id@psu.edu>... User UNknown
```

### PR DESCRIPTION
**Pull request by ajkiessl**
Adds psu_identity gem, removes PennState::SearchService code and converts implementations to PsuIdentity::SearchService.  Also, adds rake task for updating the active status for users using PsuIdentity::SearchService.

@whereismyjetpack We'll need a nightly cron job for this.  I can take a stab at it, I think I've done this before with helm charts.

Closes #843

### CODE REVIEW

**Comment by ajkiessl** [comment]
I guess I'll add some tests for the rake task since codeclimate is upset 😒

**Comment by whereismyjetpack** [comment]
I'm curious if calling this from the email job would make more sense, e.g only send the email if the user is active, mark the user inactive if they are inactive in search-service?

**Comment by ajkiessl** [comment]
I could just invoke the active status update rake task at the start of the email rake task like this:

```
namespace :user do
  desc 'Send out monthly download statistics email'
  task monthly_stats_email: :environment do
    Rails.logger.info('Starting monthly stats task...')
    Rake::Task["user:update_active_statuses"].invoke
    User.where(provider: 'azure_oauth', opt_out_stats_email: false, active: true).each do |user|
      Rails.logger.info("Putting Job on the queue for #{user.actor.email}")
      ActorMailer.with(actor: user.actor).monthly_stats.deliver_later
    end
    Rails.logger.info('Monthly stats task is complete.')
  end
```

Feels like they should at least be separate rake tasks though.

**Comment by whereismyjetpack** [comment]
I could see the `update_active_statuses` becoming a service someday, but I think calling it before running the monthly stats is the simplest path for now.

**Comment by rschenk** [comment]
Personally, I like to have rake tasks be as close to a one-liner as possible, invoking a service object and letting the service object handle almost everything. Yes, I know the current code in `lib/tasks/user.rake` does not follow this pattern, but if you look at `solr.rake` and `migration.rake` they do. I personally would advocate for that refactor, and then you don't have one rake task invoking another, you can just call the correct service object. Ultimately it's kind of the same thing, but I like the service obj approach because I find it much easier to test, especially if there are any logic branches etc

File: app/services/update_user_active_statuses.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class UpdateUserActiveStatuses
+  def self.call
```
[the remark this answers is not in the document]
Comment (ajkiessl) [reply app/services/update_user_active_statuses.rb]: I guess my refactor didn't help 🤷‍♂️

[the remark this answers is not in the document]
Comment (nicktiberi) [reply app/services/update_user_active_statuses.rb]: I wouldn't go too crazy trying to appease Codeclimate. We can ignore this if we think it's complaining unnecessarily.

[the remark this answers is not in the document]
Comment (ajkiessl) [reply app/services/update_user_active_statuses.rb]: I'd say this is ready to go then.

File: app/services/update_user_active_statuses.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class UpdateUserActiveStatuses
+  class << self
+    def call
```
[the remark this answers is not in the document]
Comment (rschenk) [reply app/services/update_user_active_statuses.rb:5]: One way you could appease Code Climate here would be to move the`rescue PsuIdentity::SearchService::NotFound` down into `def active?` and have that method return false if the error was raised. That would help with the cohesion of that method a little bit too, i.e. that private method knows about PsuIdentity stuff, and the main `call` method doesn't need to worry about it (as much)

**Review by rschenk — DISMISSED** [review]

File: app/services/update_user_active_statuses.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ajkiessl) [reply app/services/update_user_active_statuses.rb:5]: Good idea.  Just pushed these changes.

**Review by nicktiberi — APPROVED** [review]
🥇

**Pull request merged**